packages/abstract-api/src/handler.rs

```diff
@@ -1,33 +1,19 @@
-use crate::{ApiContract, ApiError};
-use abstract_sdk::{
-    base::{AbstractContract, Handler},
-    AbstractSdkError,
-};
+use crate::{state::ContractError, ApiContract};
+use abstract_sdk::base::{AbstractContract, Handler};
 use cosmwasm_std::Empty;
 
-impl<
-        Error: From<cosmwasm_std::StdError> + From<ApiError> + From<AbstractSdkError>,
-        InitMsg,
-        ExecMsg,
-        QueryMsg,
-        Receive,
-    > Handler for ApiContract<Error, InitMsg, ExecMsg, QueryMsg, Receive>
+impl<Error: ContractError, InitMsg, ExecMsg, QueryMsg, SudoMsg, Receive> Handler
+    for ApiContract<Error, InitMsg, ExecMsg, QueryMsg, SudoMsg, Receive>
 {
     type Error = Error;
-
     type CustomExecMsg = ExecMsg;
-
     type CustomInitMsg = InitMsg;
-
     type CustomQueryMsg = QueryMsg;
-
     type CustomMigrateMsg = Empty;
-
+    type SudoMsg = SudoMsg;
     type ReceiveMsg = Receive;
 
-    fn contract(
-        &self,
-    ) -> &AbstractContract<Self, Error, InitMsg, ExecMsg, QueryMsg, Empty, Receive> {
+    fn contract(&self) -> &AbstractContract<Self, Error> {
```

Comment:
Are the entry-point types inferred here?

Reply:
They are inferred through the `Handler` trait!

Reply:
see the function handler type definitions in the AbstractContract type

Reply:
Amazing. I love rust.